packages/babel-traverse/src/scope/index.js

```diff
@@ -897,10 +897,27 @@ export default class Scope {
 
   getBinding(name: string) {
     let scope = this;
+    let previousPath;
 
     do {
       const binding = scope.getOwnBinding(name);
-      if (binding) return binding;
+      if (binding) {
+        // When a pattern is a Scope, it is a part of parameter expressions.
```

Comment:
This case is one (likely the only one?) of the exception that although the `Pattern` is descended from `Function`, the scope of `Pattern` is _not_ descended from the one of `Function`. This breaks the assumption that a scope is always initialized from the scope of its nearest Scopable ancestry, which becomes `scope.parent` later.

https://github.com/babel/babel/blob/f651f22f96b2715010cd8a7b65bb12648eba8307/packages/babel-traverse/src/path/context.js#L113

I feel like it is more cumbersome to fix that on `scope.parent` accessor, so I implement the fix on `getBinding`.